### PR TITLE
Update booking app goldens to use alchemist

### PR DESCRIPTION
**Pull request by nikolay-vasilev-prime**
#### Overview

> Closes: #865

#### Checklist

*Implementation*
- [x] Implementation matches ticket acceptance criteria and technical notes
- [x] Manually tested against Acceptance Criteria
- [ ] UI checked in Light / Dark mode

*Stability*
- [x] Checked if changes affect any features and verified affected features work as expected
- [x] Added unit tests for new code and verified existing tests work as expected

*Code quality*
- [ ] Updated `CHANGELOG.md`, `README.md` and package versions in `pubspec.yaml`
- [ ] Dependencies are updated to latest versions or new tickets are created if there are breaking changes or deprecations
- [ ] If an unrelated part of the codebase needs to be updated or refactored, create tickets with proposed changes

### CODE REVIEW

File: examples/booking_app/test/feature_home/view/home_page_golden_test.dart [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
 
 void main() {
   runGoldenTests([
-    generateDeviceBuilder(
-      scenario: Scenario(name: 'home_page_search_golden_test'),
+    buildScenario(
+      scenario: 'home_page_search_golden_test',
       widget: homePageFactory(
           navigationType: Stub.searchNavItemType), //example: Stubs.emptyList
+      customPumpBeforeTest: animationCustomPump,
```
Comment (StanevPrime) [inline examples/booking_app/test/feature_home/view/home_page_golden_test.dart:11]: @GoranPrime it looks that here the animationCustomPump works as expected

File: examples/booking_app/test/feature_hotel_favorites/view/goldens/linux/dark_theme/hotel_favorites_success.png
Comment (StanevPrime) [inline examples/booking_app/test/feature_hotel_favorites/view/goldens/linux/dark_theme/hotel_favorites_success.png:1]: Do we really want to commit platform-specific golden files ?

File: examples/booking_app/test/feature_hotel_search/view/goldens/ci/dark_theme/hotel_animated_list_view_with_hotels.png
Comment (StanevPrime) [inline examples/booking_app/test/feature_hotel_search/view/goldens/ci/dark_theme/hotel_animated_list_view_with_hotels.png:1]: Why here we have a blank page ?

**Review by StanevPrime — CHANGES_REQUESTED** [review]

File: examples/booking_app/test/feature_hotel_favorites/view/goldens/linux/dark_theme/hotel_favorites_success.png
Comment (nikolay-vasilev-prime) [reply examples/booking_app/test/feature_hotel_favorites/view/goldens/linux/dark_theme/hotel_favorites_success.png:1]: This is fixed

**Review by StanevPrime — APPROVED** [review]

**Pull request merged**